lib/ruby_smb/smb2/tree.rb

```diff
@@ -127,7 +128,7 @@ def list(directory: nil, pattern: '*', type: RubySMB::Fscc::FileInformation::Fil
           # Reset the message id so the client can update appropriately.
           directory_request.smb2_header.message_id = 0
         end
-
+        opened_directory.close
```

Comment:
Does this need an ensure block, and error handling if the close fails? 🤔

Reply:
putting it in an `ensure` makes sense to me done that, error handling if the close fails uuhhhh ideally yes? but I'm not sure what you would actually want to happen there, we could retry? I wouldn't want to bomb out if we can't close it because we'd still get what we wanted which in this case is the list of files in the directory 🤷